src/ImageSharp/IO/ChunkedMemoryStream.cs

```diff
@@ -12,44 +14,23 @@ namespace SixLabors.ImageSharp.IO;
 /// Chunks are allocated by the <see cref="MemoryAllocator"/> assigned via the constructor
 /// and is designed to take advantage of buffer pooling when available.
 /// </summary>
-internal sealed class ChunkedMemoryStream : Stream
+public class ChunkedMemoryStream : Stream
 {
-    // The memory allocator.
-    private readonly MemoryAllocator allocator;
+    private readonly MemoryChunkBuffer memoryChunkBuffer;
+    private readonly byte[] singleByteBuffer = new byte[1];
 
-    // Data
-    private MemoryChunk? memoryChunk;
-
-    // The total number of allocated chunks
-    private int chunkCount;
-
-    // The length of the largest contiguous buffer that can be handled by the allocator.
-    private readonly int allocatorCapacity;
-
-    // Has the stream been disposed.
+    private long length;
+    private long position;
+    private int currentChunk;
```

Comment:
These names could be considered a little misleading/confusing.

E.g. `currentChuck` is an index into `memoryChunkBuffer`, so I think maybe call it `currentChunkIdx`.

Whereas `currentChunkIdx` is an offset/index /within/ the current chunk, so maybe call it `intraChunkByteIdx`, `chunkByteIdx`, or `checkByteOffset`? etc.

Reply:
Yeah.. good point. I've opted for `bufferIndex` and `chunkIndex` as an improvement.